pages/ekmann.js

```diff
@@ -0,0 +1,352 @@
+      },
+      body: JSON.stringify([
+        {
+          id: "1",
+          language: language,
+          text: inputText,
+        },
+      ]),
+    };
+    // Reset form after submission (optional)
+    setFormData({
+      0: "",
+      1: "",
+      2: "",
+      3: "",
+      4: "",
+      5: "",
+      6: "",
+      7: "",
+      8: "",
+      9: "",
+    });
+    try {
+      const response = await fetch(url, options);
+      const result = await response.json();
+      //.text(); makes the response a string - to read the response body as plain text
+      setEmotionResult(result);
+      setPredictionsState(true);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const question = [
+    "1. What word describes your primary emotion right now?",
+    "2. How would you describe that emotion?",
+    "3. How does your body feel?",
+    "4. Is there an event, person, place, or thing that might have caused this response?",
+    "5. How did you react to that trigger?",
+    "6. How will you respond now?",
+    "7. Are you taking care of your physical well-being?",
+    "8. What is on your mind today?",
+    "9. What can you do to feel calmer?",
+    "10. What are you grateful for today?",
+  ];
+
+  const predictions =
+    emotionResult && emotionResult[0] && emotionResult[0].predictions;
+
+  const transformedPredictions =
+    predictions &&
+    predictions[0] &&
+    predictions.map((prediction) => {
+      if (prediction.prediction === "noemo") {
+        return { ...prediction, prediction: "no-emotion" };
+      } else {
+        return prediction;
+      }
+    });
+
+  function getColorForEmotion(emotion) {
+    switch (emotion) {
+      case "joy":
+        return "yellow";
+      case "surprise":
+        return "orange";
+      case "sadness":
+        return "blue";
+      case "noemo":
+        return "gray";
+      case "fear":
+        return "purple";
+      case "anger":
+        return "red";
+      case "disgust":
+        return "green";
+      default:
+        return "lightgrey"; // Default color for unknown emotions
+    }
+  }
+
+  return (
+    <Container>
+      <Page>
+        <select id="language" onChange={handleLanguageSelect}>
+          <option value="english">english</option>
+          <option value="german">german</option>
+          <option value="espanol">espanol</option>
+        </select>
+        {page === 0 && (
+          <StyledH3>
+            Important! Please fill out all fields, otherwise the Analyzer does
+            not work!
+          </StyledH3>
+        )}
+        <p>
+          {page === 0 && question[0]}
+          {page === 1 && question[1]}
+          {page === 2 && question[2]}
+          {page === 3 && question[3]}
+          {page === 4 && question[4]}
+          {page === 5 && question[5]}
+          {page === 6 && question[6]}
+          {page === 7 && question[7]}
+          {page === 8 && question[8]}
+          {page === 9 && question[9]}
```

Comment:
Instead of repeating yourself so much here (also below with all the Textareas) you should map over the questions and render the question and the Styled Text area for each of the entries. This will ave you a lot of lines of code :) I think then you could even avoid the page state.

Reply:
`      {question[page] && question[page]}`

This is logic less code works as alternative, right now.